File: OOP_Tasks/OOP_Task6/Doctor.cpp
#pragma once
#include <iostream>
#include "Doctor.h"


Doctor::Doctor(int id, string name, string address, float salary, int patientNumber) 
	:Employee(id, name, address, salary)
{
	this->patientNumber = patientNumber;
}

Doctor::Doctor() : Employee()
{
	patientNumber = 0;
}

void Doctor::setPatientNumber(int patientNumber)
{
	this->patientNumber = patientNumber;
}

int Doctor::getPatientNumber()
{
	return (patientNumber);
}

float Doctor::addSalary(float salary)
{
	if (patientNumber >= 10)
		return (salary += 2000);
	else if (patientNumber >= 5)
		return (salary += 1000);
	else
		return (salary);
}

void Doctor::printData()
{
	cout << "Doctor id: " << id <<
	"\tDoctor name: " << name <<
	"\tDoctor address: " << address <<
	"\tPatient Number: " << patientNumber <<
	"\tDoctor salary: " << addSalary(salary) << endl;
}

File: OOP_Tasks/OOP_Task3/Doctor.cs
class Doctor:Employee
{
    int patientNumber;
    
    public Doctor(int id, string name, string address, float salary, int patientNumber):base(id, name, address, salary)
    {
        this.patientNumber = patientNumber;
    }
    
    public int patient_number
    {
        set { patientNumber = value; }
        get { return(patientNumber); }
    }
    
    public float additionSalary()
    {
        if (patientNumber > 10)
            return (salary += 2000);
        else if (patientNumber >= 5)
            return (salary += 1000);
        else
            return (salary);
    }
    
    public override string printData()
    {
        return ("doctor id: " + id + "\n" +
                "doctor name: " + name + "\n" +
                "doctor address: " + address + "\n" +
                "doctor salary: " + additionSalary() + "\n");
        
    }
}